### PR TITLE
Added gamecon_gpio_rpi driver for Raspberry Pi

### PR DESCRIPTION
**Pull request by ronj**
I have added (a fork of) the gamecon_gpio_rpi driver to Lakka. This driver enables the usage of gamepads (NES, SNES, N64, PS) directly from Raspberry Pi's GPIO pins.

There is no set-up method included at the moment, so the user needs to manually insert the kernel module with the correct arguments. I.e. in my case, having a SNES controller attached:

    # modprobe gamecon_gpio_rpi map=0,0,0,0,0,1

More information can be found on the RetroPie wiki: https://github.com/RetroPie/RetroPie-Setup/wiki/GPIO-Modules#gamecon_gpio_rpi

I have tested this on my SNES controller with built-in Pi Zero W and it works like a charm.

### CODE REVIEW

**Comment by RobLoach** [comment]
https://github.com/ronj/gamecon_gpio_rpi

Have you considered pushing your work upstream to LibreELEC? Looks pretty neat.

**Comment by kivutar** [comment]
Guys I lack time to test this. I will focus on other PRs if somebody else can take care of carefully reviewing this one.

**Comment by ronj** [comment]
@RobLoach I could try to get the driver also in LibreELEC upstream. Just FYI, I have only stripped the dkms stuff from the driver and made it possible to point to another KDIR in my fork. All the heavy lifting was already done. Credit where credits are due.

@kivutar I am using it with success for a while now, but I suppose having someone other than the author test it, is part of you process. Please let me know if I can help in some way.

**Comment by DerekRoth** [comment]
How can i test this ? Do i need to build the entire Lakka distribution from your fork ?

**Comment by kivutar** [comment]
Yes

**Comment by natinusala** [comment]
Bump, I would love to test this but I lack the hardware to do so.

I kinda want to get rid of old PRs like that one that's been sitting in a corner for a year, and start to smell in the whole room.

So, if you want to rebase it I'd happily merge it, otherwise I'll probably close it or do the changes myself if people request the feature.

**Comment by RiRilot** [comment]
I have tested this and it works perfectly.

File: Dockerfile [hunk 1]
```diff
@@ -17,30 +17,25 @@ RUN apt-get update \
 		build-essential \
 		bash \
 		bc \
-		build-essential \
```
Comment (ToKe79) [inline Dockerfile]: buitld-essential, g++, gcc and make are required, they should be kept

File: Dockerfile [hunk 2]
```diff
@@ -17,30 +17,25 @@ RUN apt-get update \
...
 		libncurses5-dev \
 		lzop \
-		make \
 		patch \
 		patchutils \
 		perl \
 		sed \
 		tar \
 		texinfo \
+		u-boot-tools \
```
Comment (ToKe79) [inline Dockerfile]: already included in master

**Pull request closed without merging**